sk0-front/src/app/auth/register/register.page.ts

```diff
@@ -1,57 +1,62 @@
 import { Component, OnInit } from '@angular/core';
 import { CommonModule } from '@angular/common';
-import { FormsModule } from '@angular/forms';
-import { IonicModule } from '@ionic/angular';
+import {
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { IonicModule, NavController } from '@ionic/angular';
 import { AlertController } from '@ionic/angular';
+import { RouterModule } from '@angular/router';
+import { AuthService } from '../auth.service';
 
 @Component({
   selector: 'app-register',
   templateUrl: './register.page.html',
   styleUrls: ['./register.page.scss'],
   standalone: true,
-  imports: [IonicModule, CommonModule, FormsModule]
+  imports: [
+    IonicModule,
+    CommonModule,
+    FormsModule,
+    ReactiveFormsModule,
+    RouterModule,
+  ],
 })
 export class RegisterPage {
-  name: string = '';
-  email: string = '';
-  password: string = '';
-
-  private readonly emailValid = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
-  private readonly passwordValid = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{6,}$/;
-
-  constructor(private alertController: AlertController) {}
+  private readonly strongPasswordValidator =
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{6,}$/;
+
+  registerForm = new FormGroup({
+    name: new FormControl('', [Validators.required]),
+    email: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', [
+      Validators.required,
+      Validators.pattern(this.strongPasswordValidator),
+    ]),
+  });
+  constructor(
+    private navCtrl: NavController,
+    private authService: AuthService
+  ) {}
 
   async onRegister() {
-    if (!this.name || !this.email || !this.password) {
-      return this.errorAlert('Minden mező kitöltése kötelező.');
-    }
+    const email = this.registerForm.get('email')!.value as string;
+    const password = this.registerForm.get('password')!.value as string;
+    const name = this.registerForm.get('name')!.value as string;
```

Comment:
I wonder if we should do this type casting here or there is a less "forceful" solution too.

Reply:
I made another solution, not sure you meant that.